### PR TITLE
darwin.stdenv: remove openssl

### PR DESCRIPTION
**Pull request by LnL7**
###### Motivation for this change

Together with #21078 this will get rid of `openssl` and `lzo` in the darwin stdenv.

I'm testing a branch [WIP](https://github.com/LnL7/nixpkgs/commits/lnl7-wip) with #21078, #21099 and #21101.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

```diff
+bootstrap-python-2.7.12.drv
+bzip2-1.0.6.0.1.drv
-bzip2-1.0.6.0.1.drv
-clang-5.3.patch
+cmake-3.6.2.drv
-cmake-3.6.2.drv
-curl-7.51.0.drv
-curl-7.51.0.tar.bz2.drv
-db-5.3.28.drv
-db-5.3.28.tar.gz.drv
-e0ec3471cb09.drv
-gdbm-1.12.drv
-gdbm-1.12.drv
-gdbm-1.12.tar.gz.drv
-hook.drv
+libarchive-3.2.2.drv
-libarchive-3.2.2.drv
-libssh2-1.7.0.drv
-libssh2-1.7.0.tar.gz.drv
-libxml2-2.9.4.drv
-link-against-ncurses.patch
+lzo-2.09.drv
-lzo-2.09.drv
-no-arch_only-6.3.patch
-openssl-1.0.2j.drv
-openssl-1.0.2j.drv
-openssl-1.0.2j.tar.gz.drv
+pkg-config-0.29.drv
-pkg-config-0.29.drv
-properly-detect-curses.patch
-python-2.7-distutils-C++.patch
-python-2.7.12.drv
-python-2.7.12.drv
-readline-6.3.tar.gz.drv
-readline-6.3p08.drv
-readline-6.3p08.drv
-readline63-001.drv
-readline63-002.drv
-readline63-003.drv
-readline63-004.drv
-readline63-005.drv
-readline63-006.drv
-readline63-007.drv
-readline63-008.drv
-setup-hook.sh
-sqlite-3.15.0.drv
-sqlite-autoconf-3150000.tar.gz.drv
-use-etc-ssl-certs.patch
```

### CODE REVIEW

**Comment by copumpkin** [comment]
Can you elaborate on what your goal here? Is it to reduce closure size? If so, how much does it improve that situation?

**Comment by LnL7** [comment]
The main thing I would like is to make diff with the linux stdenv smaller, the main things that I would like to get rid of are `curl` and `openssl`.

**Comment by copumpkin** [comment]
Ah okay!

**Comment by LnL7** [comment]
I think the main thing that brings in a lot of stuff is cmake, I'm not sure why it uses `--system-libs`.

**Comment by copumpkin** [comment]
Probably because it's nice for us to not have a bunch of different versions
of things floating around. If we need to apply a patch or fix to OpenSSL,
people might be surprised that CMake isn't using them because it bundles
its own

**Comment by LnL7** [comment]
It doesn't pull in openssl directly, it's stuff like curl and libarchive.

File: pkgs/development/libraries/libarchive/default.nix [hunk 1]
```diff
@@ -20,23 +25,32 @@ stdenv.mkDerivation rec {
...
+  '';
 
   preFixup = ''
+  '' + optionalString withLzo ''
     sed -i $lib/lib/libarchive.la \
-      -e 's|-lcrypto|-L${openssl.out}/lib -lcrypto|' \
       -e 's|-llzo2|-L${lzo}/lib -llzo2|'
+  '' + optionalString withOpenssl ''
+    sed -i $lib/lib/libarchive.la \
+      -e 's|-lcrypto|-L${openssl.out}/lib -lcrypto|
```
Comment (LnL7) [inline pkgs/development/libraries/libarchive/default.nix]: I missed a quote here.

**Comment by LnL7** [comment]
Closing in favor of #21596

**Pull request closed without merging**